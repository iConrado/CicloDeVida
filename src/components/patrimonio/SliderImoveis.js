import React from 'react';

import { 
  View, 
  Text,
  Image, } from 'react-native';
import Slider from 'react-native-slider';

import styles from '../functions/styles';
import monetizar from '../functions/monetizar';

const imgHome = require('../../imgs/ic_home_white.png');

export default class SliderImoveis extends React.Component {
  constructor(props) {
    super(props);
    this.state = { tmpImoveis: props.inicial ? props.inicial : 0 }; 
  }

  render() {
    return (
      <View style={[styles.viewVertical, styles.espacador]}>
        <Text style={styles.label}>Imóveis (valor total - financiados ou não):</Text>
        <View style={styles.viewHorizontal}>
          <View style={styles.viewIcone}>
            <Image 
              style={styles.imgIcone}
              source={imgHome}
            />
          </View>
          <View style={styles.viewPosIcone}>
            <Slider
              style={styles.patrim_slIcone}
              minimumValue={0}
              maximumValue={5000000}
              step={10000}
              minimumTrackTintColor='#14567A'
              thumbTintColor='#14567A'
              value={this.state.tmpImoveis}
              onValueChange={(value) => this.setState({ tmpImoveis: value })}
              onSlidingComplete={(value) => this.props.retorno(value)}
            />
            <Text style={styles.patrim_txIcone}>{monetizar(this.state.tmpImoveis)}</Text>
          </View>
        </View>
      </View>
    );
  }
}
