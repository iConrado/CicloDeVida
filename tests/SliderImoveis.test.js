import React from 'react';
import { shallow } from 'enzyme';
import Slider from 'react-native-slider';

import SliderImoveis from '../src/components/patrimonio/SliderImoveis';

describe('SliderImoveis', () => {
  let ret;
  const retorno = valor => {
    ret = valor;
  };
  const app = shallow(<SliderImoveis inicial={50} retorno={retorno} />);
  const app2 = shallow(<SliderImoveis retorno={retorno} />);

  it('renders correctly', () => {
    expect(app).toMatchSnapshot();
  });

  it('slider sem o valor inicial', () => {
    app2
      .find(Slider)
      .at(0)
      .simulate('valueChange', 30);
    expect(app2.state('tmpImoveis')).toBe(30);
  });

  it('zera o slider', () => {
    app
      .find(Slider)
      .at(0)
      .simulate('valueChange', 0);
    expect(app.state('tmpImoveis')).toBe(0);
  });

  it('altera o valor do slider', () => {
    app
      .find(Slider)
      .at(0)
      .simulate('valueChange', 70);
    expect(app.state('tmpImoveis')).toBe(70);
  });

  it('retorna valor do slider', () => {
    app
      .find(Slider)
      .at(0)
      .simulate('slidingComplete', 70);
    expect(ret).toBe(70);
  });
});
