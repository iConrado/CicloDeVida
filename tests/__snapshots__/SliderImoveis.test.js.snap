// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SliderImoveis renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SliderImoveis
    inicial={50}
    retorno={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#666",
            }
          }
        >
          Imóveis (valor total - financiados ou não):
        </Text>,
        <Component
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "width": "100%",
            }
          }
        >
          <Component
            style={
              Object {
                "alignContent": "center",
                "alignItems": "center",
                "backgroundColor": "#48AEB5",
                "borderRadius": 13,
                "height": 40,
                "justifyContent": "center",
                "width": 40,
              }
            }
          >
            <Image
              source={1}
              style={
                Object {
                  "height": 35,
                  "width": 35,
                }
              }
            />
          </Component>
          <Component
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "flexShrink": 1,
                "justifyContent": "space-between",
                "paddingLeft": 10,
              }
            }
          >
            <Slider
              animationType="timing"
              debugTouchArea={false}
              maximumTrackTintColor="#b3b3b3"
              maximumValue={5000000}
              minimumTrackTintColor="#14567A"
              minimumValue={0}
              onSlidingComplete={[Function]}
              onValueChange={[Function]}
              step={10000}
              style={
                Object {
                  "height": 40,
                  "width": "65%",
                }
              }
              thumbTintColor="#14567A"
              thumbTouchSize={
                Object {
                  "height": 40,
                  "width": 40,
                }
              }
              value={50}
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "height": 40,
                  "textAlign": "center",
                  "textAlignVertical": "center",
                  "width": "35%",
                }
              }
            >
              R$ 50
            </Text>
          </Component>
        </Component>,
      ],
      "style": Array [
        Object {
          "alignItems": "flex-start",
          "flexDirection": "column",
          "justifyContent": "space-between",
          "marginBottom": 8,
          "width": "100%",
        },
        Object {
          "marginTop": 15,
        },
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "Imóveis (valor total - financiados ou não):",
          "ellipsizeMode": "tail",
          "style": Object {
            "color": "#666",
          },
        },
        "ref": null,
        "rendered": "Imóveis (valor total - financiados ou não):",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Component
              style={
                Object {
                  "alignContent": "center",
                  "alignItems": "center",
                  "backgroundColor": "#48AEB5",
                  "borderRadius": 13,
                  "height": 40,
                  "justifyContent": "center",
                  "width": 40,
                }
              }
            >
              <Image
                source={1}
                style={
                  Object {
                    "height": 35,
                    "width": 35,
                  }
                }
              />
            </Component>,
            <Component
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "flexShrink": 1,
                  "justifyContent": "space-between",
                  "paddingLeft": 10,
                }
              }
            >
              <Slider
                animationType="timing"
                debugTouchArea={false}
                maximumTrackTintColor="#b3b3b3"
                maximumValue={5000000}
                minimumTrackTintColor="#14567A"
                minimumValue={0}
                onSlidingComplete={[Function]}
                onValueChange={[Function]}
                step={10000}
                style={
                  Object {
                    "height": 40,
                    "width": "65%",
                  }
                }
                thumbTintColor="#14567A"
                thumbTouchSize={
                  Object {
                    "height": 40,
                    "width": 40,
                  }
                }
                value={50}
              />
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "height": 40,
                    "textAlign": "center",
                    "textAlignVertical": "center",
                    "width": "35%",
                  }
                }
              >
                R$ 50
              </Text>
            </Component>,
          ],
          "style": Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "width": "100%",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Image
                source={1}
                style={
                  Object {
                    "height": 35,
                    "width": 35,
                  }
                }
              />,
              "style": Object {
                "alignContent": "center",
                "alignItems": "center",
                "backgroundColor": "#48AEB5",
                "borderRadius": 13,
                "height": 40,
                "justifyContent": "center",
                "width": 40,
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "source": 1,
                "style": Object {
                  "height": 35,
                  "width": 35,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Slider
                  animationType="timing"
                  debugTouchArea={false}
                  maximumTrackTintColor="#b3b3b3"
                  maximumValue={5000000}
                  minimumTrackTintColor="#14567A"
                  minimumValue={0}
                  onSlidingComplete={[Function]}
                  onValueChange={[Function]}
                  step={10000}
                  style={
                    Object {
                      "height": 40,
                      "width": "65%",
                    }
                  }
                  thumbTintColor="#14567A"
                  thumbTouchSize={
                    Object {
                      "height": 40,
                      "width": 40,
                    }
                  }
                  value={50}
                />,
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "height": 40,
                      "textAlign": "center",
                      "textAlignVertical": "center",
                      "width": "35%",
                    }
                  }
                >
                  R$ 50
                </Text>,
              ],
              "style": Object {
                "alignItems": "center",
                "flexDirection": "row",
                "flexShrink": 1,
                "justifyContent": "space-between",
                "paddingLeft": 10,
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "animationType": "timing",
                  "debugTouchArea": false,
                  "maximumTrackTintColor": "#b3b3b3",
                  "maximumValue": 5000000,
                  "minimumTrackTintColor": "#14567A",
                  "minimumValue": 0,
                  "onSlidingComplete": [Function],
                  "onValueChange": [Function],
                  "step": 10000,
                  "style": Object {
                    "height": 40,
                    "width": "65%",
                  },
                  "thumbTintColor": "#14567A",
                  "thumbTouchSize": Object {
                    "height": 40,
                    "width": 40,
                  },
                  "value": 50,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "accessible": true,
                  "allowFontScaling": true,
                  "children": "R$ 50",
                  "ellipsizeMode": "tail",
                  "style": Object {
                    "height": 40,
                    "textAlign": "center",
                    "textAlignVertical": "center",
                    "width": "35%",
                  },
                },
                "ref": null,
                "rendered": "R$ 50",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#666",
              }
            }
          >
            Imóveis (valor total - financiados ou não):
          </Text>,
          <Component
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "width": "100%",
              }
            }
          >
            <Component
              style={
                Object {
                  "alignContent": "center",
                  "alignItems": "center",
                  "backgroundColor": "#48AEB5",
                  "borderRadius": 13,
                  "height": 40,
                  "justifyContent": "center",
                  "width": 40,
                }
              }
            >
              <Image
                source={1}
                style={
                  Object {
                    "height": 35,
                    "width": 35,
                  }
                }
              />
            </Component>
            <Component
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "flexShrink": 1,
                  "justifyContent": "space-between",
                  "paddingLeft": 10,
                }
              }
            >
              <Slider
                animationType="timing"
                debugTouchArea={false}
                maximumTrackTintColor="#b3b3b3"
                maximumValue={5000000}
                minimumTrackTintColor="#14567A"
                minimumValue={0}
                onSlidingComplete={[Function]}
                onValueChange={[Function]}
                step={10000}
                style={
                  Object {
                    "height": 40,
                    "width": "65%",
                  }
                }
                thumbTintColor="#14567A"
                thumbTouchSize={
                  Object {
                    "height": 40,
                    "width": 40,
                  }
                }
                value={50}
              />
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "height": 40,
                    "textAlign": "center",
                    "textAlignVertical": "center",
                    "width": "35%",
                  }
                }
              >
                R$ 50
              </Text>
            </Component>
          </Component>,
        ],
        "style": Array [
          Object {
            "alignItems": "flex-start",
            "flexDirection": "column",
            "justifyContent": "space-between",
            "marginBottom": 8,
            "width": "100%",
          },
          Object {
            "marginTop": 15,
          },
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "Imóveis (valor total - financiados ou não):",
            "ellipsizeMode": "tail",
            "style": Object {
              "color": "#666",
            },
          },
          "ref": null,
          "rendered": "Imóveis (valor total - financiados ou não):",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Component
                style={
                  Object {
                    "alignContent": "center",
                    "alignItems": "center",
                    "backgroundColor": "#48AEB5",
                    "borderRadius": 13,
                    "height": 40,
                    "justifyContent": "center",
                    "width": 40,
                  }
                }
              >
                <Image
                  source={1}
                  style={
                    Object {
                      "height": 35,
                      "width": 35,
                    }
                  }
                />
              </Component>,
              <Component
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "flexShrink": 1,
                    "justifyContent": "space-between",
                    "paddingLeft": 10,
                  }
                }
              >
                <Slider
                  animationType="timing"
                  debugTouchArea={false}
                  maximumTrackTintColor="#b3b3b3"
                  maximumValue={5000000}
                  minimumTrackTintColor="#14567A"
                  minimumValue={0}
                  onSlidingComplete={[Function]}
                  onValueChange={[Function]}
                  step={10000}
                  style={
                    Object {
                      "height": 40,
                      "width": "65%",
                    }
                  }
                  thumbTintColor="#14567A"
                  thumbTouchSize={
                    Object {
                      "height": 40,
                      "width": 40,
                    }
                  }
                  value={50}
                />
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "height": 40,
                      "textAlign": "center",
                      "textAlignVertical": "center",
                      "width": "35%",
                    }
                  }
                >
                  R$ 50
                </Text>
              </Component>,
            ],
            "style": Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "width": "100%",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Image
                  source={1}
                  style={
                    Object {
                      "height": 35,
                      "width": 35,
                    }
                  }
                />,
                "style": Object {
                  "alignContent": "center",
                  "alignItems": "center",
                  "backgroundColor": "#48AEB5",
                  "borderRadius": 13,
                  "height": 40,
                  "justifyContent": "center",
                  "width": 40,
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "source": 1,
                  "style": Object {
                    "height": 35,
                    "width": 35,
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Slider
                    animationType="timing"
                    debugTouchArea={false}
                    maximumTrackTintColor="#b3b3b3"
                    maximumValue={5000000}
                    minimumTrackTintColor="#14567A"
                    minimumValue={0}
                    onSlidingComplete={[Function]}
                    onValueChange={[Function]}
                    step={10000}
                    style={
                      Object {
                        "height": 40,
                        "width": "65%",
                      }
                    }
                    thumbTintColor="#14567A"
                    thumbTouchSize={
                      Object {
                        "height": 40,
                        "width": 40,
                      }
                    }
                    value={50}
                  />,
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "height": 40,
                        "textAlign": "center",
                        "textAlignVertical": "center",
                        "width": "35%",
                      }
                    }
                  >
                    R$ 50
                  </Text>,
                ],
                "style": Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "flexShrink": 1,
                  "justifyContent": "space-between",
                  "paddingLeft": 10,
                },
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "animationType": "timing",
                    "debugTouchArea": false,
                    "maximumTrackTintColor": "#b3b3b3",
                    "maximumValue": 5000000,
                    "minimumTrackTintColor": "#14567A",
                    "minimumValue": 0,
                    "onSlidingComplete": [Function],
                    "onValueChange": [Function],
                    "step": 10000,
                    "style": Object {
                      "height": 40,
                      "width": "65%",
                    },
                    "thumbTintColor": "#14567A",
                    "thumbTouchSize": Object {
                      "height": 40,
                      "width": 40,
                    },
                    "value": 50,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "accessible": true,
                    "allowFontScaling": true,
                    "children": "R$ 50",
                    "ellipsizeMode": "tail",
                    "style": Object {
                      "height": 40,
                      "textAlign": "center",
                      "textAlignVertical": "center",
                      "width": "35%",
                    },
                  },
                  "ref": null,
                  "rendered": "R$ 50",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
